experimental/PlaneCurve/EllCurveZnZ.jl

```diff
@@ -0,0 +1,704 @@
+export sum_Point_EllCurveZnZ,
+    ECM,
+    rand_pair_EllCurve_Point,
+    IntMult_Point_EllCurveZnZ,
+    cornacchia_algorithm,
+    Miller_Rabin_test,
+    Pollard_rho,
+    Pollard_p_1,
+    ECPP
+
+################################################################################
+# Elliptic curves over a ring Z/nZ
+################################################################################
+################################################################################
+# Helping functions
+################################################################################
+# addition on two projective points on an elliptic curve over Z/nZ, given in
+# short Weierstrass form, when it is possible. It gives as an output a pair
+# composed of the coordinates of the sum and ZZ(1) if the sum exists, and
+# (0, 0, 0) and the gcd otherwise.
+# Adapted from the corresponding code in Hecke.
+
+function _add(
+    P::Array{Nemo.fmpz_mod,1},
+    Q::Array{Nemo.fmpz_mod,1},
+    E::Array{Nemo.fmpz_mod,1},
+)
+    length(P) == 3 && length(Q) == 3 || error("arrays of size 3 required")
+    length(E) == 2 || error("array of size 2 required")
+    A = P[1].parent
```

Comment:
In general, avoid writing `obj.parent` instead use `parent(obj)`
```suggestion
    A = parent(P[1])
```

Reply:
I changed it where it was possible but there is no parent function for ProjSpcElem.